### PR TITLE
Add extra metadata fields

### PR DESCRIPTION
**Pull request by Jin-Sun-tts**
# Pull Request

Related to https://github.com/GSA/data.gov/issues/4847

## About

Added three keys and their corresponding values to the extras field in the metadata.
Replaced org_id with harvest_source_id when clearing the dataset as it is available now. 
Added new route for raw record data `harvest_record/<record_id>/raw
`

### CODE REVIEW

File: harvester/harvest.py [hunk 1]
```diff
@@ -515,7 +514,7 @@ def update_self_in_db(self) -> bool:
     def ckanify_dcatus(self) -> None:
         try:
             self.ckanified_metadata = ckanify_dcatus(
-                self.metadata, self.harvest_source.organization_id
+                self.metadata, self.harvest_source
```
Comment (FuhuXia) [inline harvester/harvest.py:518]: `self.harvest_source.organization_id` and `self.harvest_source` are different types, how does one replace the other? could it be `self.harvest_source.id`?

File: harvester/utils/ckan_utils.py [hunk 2]
```diff
@@ -151,7 +151,7 @@ def munge_tag(tag: str) -> str:
     return tag
 
 
-def create_ckan_extras(metadata: dict) -> list[dict]:
+def create_ckan_extras(metadata: dict, harvest_source) -> list[dict]:
```
Comment (FuhuXia) [inline harvester/utils/ckan_utils.py]: provide type hint to be consistent.

File: harvester/utils/ckan_utils.py [hunk 3]
```diff
@@ -283,14 +301,14 @@ def simple_transform(metadata: dict, owner_org: str) -> dict:
     return output
 
 
-def ckanify_dcatus(metadata: dict, owner_org: str) -> dict:
-    ckanified_metadata = simple_transform(metadata, owner_org)
+def ckanify_dcatus(metadata: dict, harvest_source) -> dict:
```
Comment (FuhuXia) [inline harvester/utils/ckan_utils.py]: provide type hint to be consistent.

**Review by FuhuXia — CHANGES_REQUESTED** [review]

File: harvester/harvest.py (hunk 1 above)
Comment (Jin-Sun-tts) [reply harvester/harvest.py:518]: Previously, we only required the org_id, but now we need additional fields from harvest_source. As a result, we are passing the entire object to allow access to the other fields from harvest_source.

File: harvester/utils/ckan_utils.py (hunk 2 above)
Comment (Jin-Sun-tts) [reply harvester/utils/ckan_utils.py]: Added.

File: harvester/utils/ckan_utils.py (hunk 3 above)
Comment (Jin-Sun-tts) [reply harvester/utils/ckan_utils.py]: The HarvestSource class is defined in harvest.py, which imports functions from ckan_util. This results in a circular import error. To resolve this, move the import of the function to the specific location where it is used. Then added the type hint into this function.

**Review by FuhuXia — APPROVED** [review]

**Pull request merged**